### PR TITLE
Simplify/improve FileCache low read-rate logic

### PR DESCRIPTION
**Pull request by arnova**
This simplifies the logic that determines low read-rate conditions. It also gets rid of resetting the filling-flag when level reaches 0 as in that case we (also) terminate making that operation useless.

### CODE REVIEW

File: xbmc/filesystem/FileCache.cpp [hunk 1]
```diff
@@ -114,14 +114,14 @@ bool CFileCache::Open(const CURL& url)
 
   CSingleLock lock(m_sync);
 
-  CLog::Log(LOGDEBUG,"CFileCache::Open - opening <%s> using cache", url.GetFileName().c_str());
+  m_sourcePath = url.GetRedacted();
 
-  m_sourcePath = url.Get();
+  CLog::Log(LOGDEBUG,"CFileCache::Open - opening <%s>", m_sourcePath.c_str());
```
Comment (wsnipex) [inline xbmc/filesystem/FileCache.cpp:119]:
```suggestion
  CLog::Log(LOGDEBUG,"{} - opening <{}>", __FUNCTION__, m_sourcePath);
```

File: xbmc/filesystem/FileCache.cpp [hunk 2]
```diff
@@ -114,14 +114,14 @@ bool CFileCache::Open(const CURL& url)
...
 
   // opening the source file.
-  if (!m_source.Open(m_sourcePath, READ_NO_CACHE | READ_TRUNCATED | READ_CHUNKED))
+  if (!m_source.Open(url.Get(), READ_NO_CACHE | READ_TRUNCATED | READ_CHUNKED))
   {
-    CLog::Log(LOGERROR,"%s - failed to open source <%s>", __FUNCTION__, url.GetRedacted().c_str());
+    CLog::Log(LOGERROR,"%s - <%s> failed to open source", __FUNCTION__, m_sourcePath.c_str());
```
Comment (wsnipex) [inline xbmc/filesystem/FileCache.cpp:124]:
```suggestion
    CLog::Log(LOGERROR,"{} - <{}> failed to open source", __FUNCTION__, m_sourcePath);
```

File: xbmc/filesystem/FileCache.cpp [hunk 3]
```diff
@@ -138,8 +138,8 @@ bool CFileCache::Open(const CURL& url)
   m_chunkSize = CFile::DetermineChunkSize(
       m_source.GetChunkSize(),
       CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_cacheChunkSize);
-  CLog::Log(LOGDEBUG, "CFileCache::Open - Source chunk size is %i, setting cache chunk size to %i",
-            m_source.GetChunkSize(), m_chunkSize);
+  CLog::Log(LOGDEBUG, "CFileCache::Open - <%s> Source chunk size is %i, setting cache chunk size to %i",
```
Comment (wsnipex) [inline xbmc/filesystem/FileCache.cpp:141]:
```suggestion
  CLog::Log(LOGDEBUG, "{} - <{}> Source chunk size is {}, setting cache chunk size to {}",
            __FUNCTION__, m_sourcePath, m_source.GetChunkSize(), m_chunkSize);
```

**Review by wsnipex — APPROVED** [review]
Looks good to me, but please consider changing to libfmt log format while you touch this

**Comment by arnova** [comment]
> Looks good to me, but please consider changing to libfmt log format while you touch this

Sure, no problem. Thanks for the pointer.

@DaveTBlake : Do you accept the logging (only) changes for Matrix? Would be nice to have this in as it would make debugging so much easier. If so, I'll split off the rest of the changes.

**Comment by DaveTBlake** [comment]
> Do you accept the logging (only) changes for Matrix? Would be nice to have this in as it would make debugging so much easier. If so, I'll split off the rest of the changes.

Looking to freeze for release this weekend @arnova so trying to avoid the "one more small thing" situation unless it is vital or we get delayed by other factors. But other than that happy to have improved logging for debugging etc. and would put it into 19.1 if it misses 19.0. So really up to you and how fast you can be.

**Comment by arnova** [comment]
> > Do you accept the logging (only) changes for Matrix? Would be nice to have this in as it would make debugging so much easier. If so, I'll split off the rest of the changes.
> 
> Looking to freeze for release this weekend @arnova so trying to avoid the "one more small thing" situation unless it is vital or we get delayed by other factors. But other than that happy to have improved logging for debugging etc. and would put it into 19.1 if it misses 19.0. So really up to you and how fast you can be.

See https://github.com/xbmc/xbmc/pull/19218 . It's up to you if you want to merge it for 19.0 or 19.1. It *could* help a lot if more problems with curl/cache surface like the http2 issues we're seeing.

**Pull request merged**